cmd/object-handlers.go

```diff
@@ -248,22 +263,41 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 			Expression:           selectReq.Expression,
 			OutputFieldDelimiter: selectReq.OutputSerialization.CSV.FieldDelimiter,
 			StreamSize:           objInfo.Size,
-			HeaderOpt:            selectReq.InputSerialization.CSV.FileHeaderInfo == CSVFileHeaderInfoUse,
+			HeaderOpt:            selectReq.InputSerialization.CSV.FileHeaderInfo == s3select.CSVFileHeaderInfoUse,
 			Progress:             selectReq.RequestProgress.Enabled,
 		}
-		s3s, err := s3select.NewInput(options)
+		s3s, err = s3select.NewInput(options)
 		if err != nil {
 			writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 			return
 		}
-		_, _, _, _, _, _, err = s3s.ParseSelect(selectReq.Expression)
+
+	}
+
+	if selectReq.InputSerialization.JSON != nil {
+		options := &s3select.JSONOptions{
+			Name:       "S3Object", // Default table name for all objects
+			ReadFrom:   br,
+			Compressed: string(selectReq.InputSerialization.CompressionType),
+			Expression: selectReq.Expression,
+			StreamSize: objInfo.Size,
+			Type:       selectReq.InputSerialization.JSON.Type == s3select.JSONTypeDocument,
+			Progress:   selectReq.RequestProgress.Enabled,
+		}
+		s3s, err = s3select.NewJSONInput(options)
 		if err != nil {
 			writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 			return
 		}
-		if err = s3s.Execute(w); err != nil {
-			logger.LogIf(ctx, err)
-		}
+	}
+	_, _, _, _, _, _, err = s3select.ParseSelect(selectReq.Expression, s3s)
```

Comment:
Are the return values of `ParseSelect` used anywhere if not, you can just return `err`

Reply:
In ```object-handlers.go``` only the error is checked  but in ```select.go``` all the returned values are used.

Reply:
ok